fuse/handle.go

```diff
@@ -113,32 +65,47 @@ const maxInt = int(^uint(0) >> 1)
 
 // Write is called to write a block of data at a certain offset.
 func (hd *Handle) Write(ctx context.Context, req *fuse.WriteRequest, resp *fuse.WriteResponse) error {
+	start := time.Now()
+	// Note: even when underlying process makes consequent writes,
+	// the kernel might send blocks out of order!!!
```

Comment:
> Can you elaborate a bit on this? Especially what the consequences for us are.

I will add the comment but the full description is the following:
let's say at the OS level you do `(cat or dd) largeFile > /brig-fuse-moung/file`. You would naively expect `Write` request to be in `fifo` order with steady growing offsets (I called it consequent), and they mostly are.  What I saw that occasionally OS layer asks to write a block with higher offset and then another with a lower offset.

I noticed this when I was tracking offset position: with `fifo` order I would not have to do additional `Seek` if position matches offset.

It does not concern our current `Write` much. But one should not assume that `Write` could be streamed with automatically positioned cursor.
Otherwise, it would be super easy to merge `overlays` to one big `modification` which might help with speed. 

But it also makes, aforementioned swapping to disk hard 'since overlay might have a hole'. So every block would be of a different size.

Reply:
addressed in 03e48e64